helix-core/src/syntax.rs

```diff
@@ -706,13 +709,25 @@ thread_local! {
     })
 }
 
-#[derive(Debug)]
 pub struct Syntax {
     layers: HopSlotMap<LayerId, LanguageLayer>,
+    layers_lut: RawTable<LayerId>,
+    layers_lut_hasher: RandomState,
```

Comment:
I think this may be clearer as `layer_table` and `layer_hasher`. Both may need doc comments too to point out what they're used for.

I'm afraid of abbreviations :P

Reply:
I changed the name and added some doc comments as suggested.
In the process I also found that I can use `PartialEq` and `Hash` directly instead of a freestanding function which looks more similar to a traditional `HashMap` and might be easier to understand.